Src/GameConnection.hpp

```diff
@@ -3,15 +3,52 @@
 #include <memory>
 #include <vector>
 #include "Object.hpp"
-
+#include "Windows.hpp"
+enum class PlayerState:int
+{
+	NOTCREATED=0,
+	NORMAL=1,
+	INVINCIBLE =2,
+	DYING=4
+};
 struct GameConnection {
     virtual ~GameConnection() noexcept = default;
-    virtual void GetPowers(std::vector<Object> &powers) noexcept = 0;
-    virtual void GetEnemyData(std::vector<Object> &enemy) noexcept = 0;
-    virtual void GetEnemyBulletData(std::vector<Object> &bullet, const Player &player, double maxRange) noexcept = 0;
-    virtual void GetPlayerData(Player &self) noexcept = 0;
-    virtual void GetEnemyLaserData(std::vector<Laser> &laser) noexcept = 0;
+	virtual int getTimeline() noexcept = 0;
+    virtual void getPowers(std::vector<Object> &powers) noexcept = 0;
+    virtual void getEnemyData(std::vector<Object> &enemy) noexcept = 0;
+    virtual void getEnemyBulletData(std::vector<Object> &bullet, const Player &player, double maxRange) noexcept = 0;
+    virtual void getPlayerData(Player &self) noexcept = 0;
+    virtual void getEnemyLaserData(std::vector<Laser> &laser) noexcept = 0;
     virtual void sendKeyInfo(int dir,bool shift,bool z,bool x) noexcept = 0;
+	virtual void getMousePosition(Vec2d& pos)noexcept = 0;
+	virtual PlayerState GetPlayerStateInformation()noexcept = 0;
+};
+class GameConnectionTH10 : public GameConnection {
+public:
+	GameConnectionTH10();
+	~GameConnectionTH10() noexcept override;
+	void getPowers(std::vector<Object> &powers) noexcept override;
+	void getEnemyData(std::vector<Object> &enemy) noexcept override;
+	void getEnemyBulletData(std::vector<Object> &bullet, const Player &player, double maxRange) noexcept override;
+	void getPlayerData(Player &self) noexcept override;
+	void getEnemyLaserData(std::vector<Laser> &laser) noexcept override;
+	void sendKeyInfo(int dir, bool shift, bool z, bool x) noexcept override;
+	void getMousePosition(Vec2d& pos)noexcept override;
+	int getTimeline() noexcept override;
+	PlayerState GetPlayerStateInformation()noexcept override;
```

Comment:
Please apply naming conventions consistently if you decide to make any changes to that